cpp/include/cudf/functions.h

```diff
@@ -796,6 +796,42 @@ gdf_error gdf_min(gdf_column *col, void *dev_result, gdf_size_type dev_result_si
  * --------------------------------------------------------------------------*/
 gdf_error gdf_max(gdf_column *col, void *dev_result, gdf_size_type dev_result_size);
 
+/* --------------------------------------------------------------------------*
+ * @brief  Computes the rolling window function of the values in a column.
+ *
+ * This matches Pandas' API for DataFrame.rolling with a couple notable
+ * differences:
+ * - instead of the center flag it uses the forward window size to allow for
+ * more flexible windows. The total window size = window + forward_window.
+ * Element i uses elements [i-window+1, i+forward_window] to do the window
+ * computation.
+ * - instead of storing NA/NaN for output rows that do not meet the minimum
+ * number of observations this function updates the valid bitmask of the column
+ * to indicate which elements are valid.
+ *
+ * @param[out] output_col The output column
+ * @param[in] input_col The input column
+ * @param[in] window The rolling window size, output_col[i] accumulates values
+ *                from input_col[i-window+1] to input_col[i] inclusive
+ * @param[in] min_periods Minimum number of observations in window required to
+ *                have a value, otherwise 0 is stored in the valid bit mask for
+ *                the element i
+ * @param[in] forward_window The window size in the forward direction,
+ *                output_col[i] accumulates values from input_col[i] to
+ *                input_col[i+forward_window] inclusive
+ * @param[in] agg_type The rolling window aggregtion type (sum, max, min, etc.)
+ *
+ * @return    GDF_SUCCESS if the operation was successful, otherwise an
+ *            appropriate error code.
+ *
+ * --------------------------------------------------------------------------*/
+gdf_error gdf_rolling_window(gdf_column *output_col,
+			     const gdf_column *input_col,
```

Comment:
Usually rolling operations are done in groups as opposed to a column at a time.

Reply:
I guess in pandas it might be different but windows are often combined in many data operations across multiple values

Reply:
Even if windows are executed across multiple values, these are independent operations and `gdf_rolling_window` can be called multiple times. I do see some value in providing an API that can handle multiple columns, which can benefit kernel launch timings, but otherwise there is no reuse of data between the columns. To avoid complicating the initial API, I would postpone this feature to the next version of the API, or until we get a realistic case that suffers from kernel launch latencies for rolling windows. Sounds good?

Reply:
yes this sounds fine